### PR TITLE
Added tests for sliding time windows

### PR DESCRIPTION
**Pull request by Arthasasyan**
Pending for 6560#note-11 to be resolved

### CODE REVIEW

**Comment by Arthasasyan** [comment]
TODO: add all functions to test

File: src/test/java/com/axibase/tsd/api/method/sql/function/aggregate/SlidingTimeWindowTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        String sqlQuery = String.format("SELECT value FROM \"%s\" WITH ROW_NUMBER(entity ORDER BY time) BETWEEN 1 MINUTE PRECEDING AND 4 MINUTE PRECEDING", METRIC);
+        String errorMessage = "Syntax error at line 1 position 178: no viable alternative at input 'WITH ROW_NUMBER(entity ORDER BY time) BETWEEN 1 MINUTE PRECEDING AND 4'";
+        assertBadRequest("Request with ROW_NUMBER returned unexpected error", errorMessage, sqlQuery);
+    }
+
+    @Test(description = "Tests only one row is affected by aggregation if ROW_NUMBER interval is less than interval between values")
+    @Issue("6560")
+    public void testAggregationWithLowerInterval() {
+        String sqlQuery = String.format("SELECT value, COUNT(VALUE) FROM \"%s\" WITH ROW_NUMBER(entity ORDER BY time) BETWEEN 1 MINUTE PRECEDING AND CURRENT ROW", METRIC);
+        List<List<String>> expectedResult = Arrays.stream(DATA)
```
[the remark this answers is not in the document]
Comment (raipc) [reply src/test/java/com/axibase/tsd/api/method/sql/function/aggregate/SlidingTimeWindowTest.java]: Actually all blocks of code that generate expectedResult can be extracted to a method `prepareExpectedResult(IntFunction<String> aggregationResultProvider)`. Function's parameters must be row number, not sample value

File: src/test/java/com/axibase/tsd/api/method/sql/function/aggregate/SlidingTimeWindowTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        assertSqlQueryRows(expectedResult, sqlQuery);
+    }
+
+    @Test(description = "Tests only two rows are affected by aggregation if ROW_NUMBER interval is more than interval between two values")
+    @Issue("6560")
+    public void testAggregationWithTwoRowsAffected() {
+        String sqlQuery = String.format("SELECT value, COUNT(VALUE) FROM \"%s\" WITH ROW_NUMBER(entity ORDER BY time) BETWEEN 2 HOUR PRECEDING AND CURRENT ROW", METRIC);
+        List<List<String>> expectedResult = new ArrayList<>();
+        for(int i = 0; i < DATA.length; i++) {
+            List<String> row;
```
[the remark this answers is not in the document]
Comment (Arthasasyan) [reply src/test/java/com/axibase/tsd/api/method/sql/function/aggregate/SlidingTimeWindowTest.java]: I suppose, this is a better option:

```java
for(int i = 0; i < DATA.length; i++) {
            List<String> row = Arrays.asList(String.valueOf(DATA[i]), (i == 0) ? "1" : "2");
            expectedResult.add(row);
        }
```

**Comment by Arthasasyan** [comment]
Now all functions take row number as argument

**Pull request merged**